Add a flag that allows breaking on NumPy ops

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107689
* #107746
* #107711
* #107710
* #107688
* __->__ #107687

This was removed in https://github.com/pytorch/pytorch/pull/106211/commits/63d406a6a9502a478eda5cf17601e95e6e896ce3
Resotiring, as it's rather useful for debugging.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

lezcano committed:
Add a flag that allows breaking on NumPy ops

This was removed in https://github.com/pytorch/pytorch/pull/106211/commits/63d406a6a9502a478eda5cf17601e95e6e896ce3
Resotiring, as it's rather useful for debugging.

[ghstack-poisoned]

lezcano committed:
Update on "Add a flag that allows breaking on NumPy ops"


This was removed in https://github.com/pytorch/pytorch/pull/106211/commits/63d406a6a9502a478eda5cf17601e95e6e896ce3
Resotiring, as it's rather useful for debugging.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

larryliu0820 commented on torch/_dynamo/config.py:
```diff
@@ -251,6 +251,9 @@
 )
 
+# Trace through NumPy or graphbreak
+trace_numpy = True
```
We had this flag before and it's deleted when upstreaming numpy_torch_interop. Why do we want to add it back?

lezcano replied:
As per the op:
>This was removed in https://github.com/pytorch/pytorch/commit/63d406a6a9502a478eda5cf17601e95e6e896ce3
Restoring, as it's rather useful for debugging.

lezcano committed:
Update on "Add a flag that allows breaking on NumPy ops"


This was removed in https://github.com/pytorch/pytorch/pull/106211/commits/63d406a6a9502a478eda5cf17601e95e6e896ce3
Resotiring, as it's rather useful for debugging.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

lezcano committed:
Update on "Add a flag that allows breaking on NumPy ops"


This was removed in https://github.com/pytorch/pytorch/pull/106211/commits/63d406a6a9502a478eda5cf17601e95e6e896ce3
Resotiring, as it's rather useful for debugging.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

lezcano committed:
Update on "Add a flag that allows breaking on NumPy ops"


This was removed in https://github.com/pytorch/pytorch/pull/106211/commits/63d406a6a9502a478eda5cf17601e95e6e896ce3
Resotiring, as it's rather useful for debugging.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]
